Mathlib/Data/BitVec/Defs.lean

```diff
@@ -0,0 +1,104 @@
+/-
+Copyright (c) 2023 Wojciech Nawrocki. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Harun Khan, Alex Keizer, Abdalrhman M Mohamed
+-/
+
+import Mathlib.Data.Fin.Basic
+import Mathlib.Data.Nat.Bitwise
+import Mathlib.Data.ZMod.Defs
+import Std.Data.BitVec
+
+
+/-!
+# Basic operations on bitvectors
+Std has defined bitvector of length `w` as `Fin (2^w)`.
+Here we define a few more operations on these bitvectors
+
+## Main definitions
+
+* `Std.BitVec.sgt`: Signed greater-than comparison of bitvectors
+* `Std.BitVec.sge`: Signed greater-equals comparison of bitvectors
+* `Std.BitVec.ugt`: Unsigned greater-than comparison of bitvectors
+* `Std.BitVec.uge`: Unsigned greater-equals comparison of bitvectors
+* `Std.BitVec.ofLEList`: Turn a little-endian list of Booleans into a bitvector
+* `Std.BitVec.ofBEList`: Turn a big-endian list of Booleans into a bitvector
+
+-/
+
+variable {w v : Nat}
+
+namespace Std.BitVec
+
+#align bitvec Std.BitVec
+#align bitvec.zero Std.BitVec.zero
+#align bitvec.cong Std.BitVec.cast
+#align bitvec.append Std.BitVec.append
+#align bitvec.shl Std.BitVec.shiftLeft
+#align bitvec.ushr Std.BitVec.ushiftRight
+#align bitvec.sshr Std.BitVec.sshiftRight
+#align bitvec.not Std.BitVec.not
+#align bitvec.and Std.BitVec.and
+#align bitvec.or Std.BitVec.or
+#align bitvec.xor Std.BitVec.xor
+#align bitvec.neg Std.BitVec.neg
+#align bitvec.add Std.BitVec.add
+#align bitvec.sub Std.BitVec.sub
+#align bitvec.mul Std.BitVec.mul
+#align bitvec.ult Std.BitVec.ult
+#align bitvec.ule Std.BitVec.ule
+#align bitvec.slt Std.BitVec.slt
+#align bitvec.sle Std.BitVec.sle
+#align bitvec.of_nat Std.BitVec.ofNat
+#align bitvec.add_lsb Nat.bit
+#align bitvec.to_nat Std.BitVec.toNat
+#align bitvec.of_fin Std.BitVec.ofFin
+#align bitvec.to_fin Std.BitVec.toFin
+#align bitvec.to_int Std.BitVec.toInt
+
+#noalign bitvec.one
+#noalign bitvec.adc
+#noalign bitvec.sbb
+#noalign bitvec.uborrow
+#noalign bitvec.sborrow
```

Comment:
Sure, but would it be OK to do this in another PR? I currently have a second PR ready that gives shows that `BitVec` is a commutative ring, but it's blocked on the current PR. If we merge this one, then we can work on bringing back these definitions in parallel to the Ring-PR being reviewed

Reply:
I think the convention is that mathlib never gets worse. We shouldn't lose things and get them back, just in case somebody needs them and they get annoyed.

Reply:
Turns out that `sborrow` and `uborrow` are just `slt` and `ult`, so I aligned them to that.
I've also added `adc` and `sbb`, although I would personally classify those as auxiliary defs, and not particularly interesting by themselves. For context, this is their implementation
```lean
/-- Add with carry (no overflow) -/
def adc {n} (x y : BitVec n) (c : Bool) : BitVec (n+1) :=
  ofFin (x.toNat + y.toNat + c.toNat)
#align bitvec.adc Std.BitVec.adc

/-- Subtract with borrow -/
def sbb {n} (x y : BitVec n) (b : Bool) : Bool × BitVec n :=
  let y := y + ofFin b.toNat
  (x < y, x - y)
#align bitvec.sbb Std.BitVec.sbb
```